### PR TITLE
Modernize http-api - move from `browser-request` to `fetch`

### LINKED ISSUE #2415: TypeError: this.opts.request is not a function when running client in Node.js; reopen issue #1191
I would like to reopen the issue #1191 since there is a similar error.
I've installed ```matrix-sdk-js``` using ```Node.js v16.15.0``` and initiated client as per example. Received ```TypeError: this.opts.request is not a function```. Also tried with nodejs v12, ```import``` or ```require``` of SDK, using ```async ... await``` function, but the same issue.
any ideas?

_Originally posted by @jt0in3e in https://github.com/matrix-org/matrix-js-sdk/issues/1191#issuecomment-1141020280_

### PR DESCRIPTION
**Pull request by t3chguy**
Fixes https://github.com/matrix-org/matrix-js-sdk/issues/2415
Fixes https://github.com/matrix-org/matrix-js-sdk/issues/801

Notes: Changes the `uploadContent` API, kills off `request` and `browser-request` in favour of `fetch`, removed callback support on a lot of the methods, adds a lot of tests.

---
Here's what your changelog entry will look like:

## 🚨 BREAKING CHANGES
 * Changes the `uploadContent` API, kills off `request` and `browser-request` in favour of `fetch`, removed callback support on a lot of the methods, adds a lot of tests. ([\#2719](https://github.com/matrix-org/matrix-js-sdk/pull/2719)). Fixes #2415 and #801.

### CODE REVIEW

File: src/http-api/prefix.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    V1 = "/_matrix/identity/api/v1",
+    /**
+     * URI path for the v2 identity API
+     */
+    V2 = "/_matrix/identity/api/v2",
+}
+
+export enum MediaPrefix {
+    /**
+     * URI path for the media repo API
```
Comment (turt2live) [inline src/http-api/prefix.ts:50]: Media exists on `/v3` now, ftr.

Comment (t3chguy) [reply src/http-api/prefix.ts:50]: Lets leave that to a separate PR

Comment (turt2live) [reply src/http-api/prefix.ts:50]: can we at least update the comment to say it's the "release 0 of the Media API" or something?

**Review by turt2live — APPROVED** [review]
I've skimmed this and it seems fine enough - not worth blocking this much longer given the chance of conflicts.

If there's something disagreeable code-wise in here, it can be fixed in a new PR.

Thanks for taking this on - it looks like an absolute epic :)

**Comment by tiennguyen1293** [comment]
i'm facing an issue with `global.__js_sdk_entrypoint` when i run the unit test. Please help take a look.
image

**Pull request merged**